admin/section/class-convertkit-settings-tools.php

```diff
@@ -0,0 +1,428 @@
+
+		$return  = apply_filters( 'convertkit_sysinfo_after_php_config', $return );
+
+		// PHP extensions and such
+		$return .= "\n" . '-- PHP Extensions' . "\n\n";
+		$return .= 'cURL:                     ' . ( function_exists( 'curl_init' ) ? 'Supported' : 'Not Supported' ) . "\n";
+		$return .= 'fsockopen:                ' . ( function_exists( 'fsockopen' ) ? 'Supported' : 'Not Supported' ) . "\n";
+		$return .= 'SOAP Client:              ' . ( class_exists( 'SoapClient' ) ? 'Installed' : 'Not Installed' ) . "\n";
+		$return .= 'Suhosin:                  ' . ( extension_loaded( 'suhosin' ) ? 'Installed' : 'Not Installed' ) . "\n";
+
+		$return  = apply_filters( 'convertkit_sysinfo_after_php_ext', $return );
+
+		// Session stuff
+		$return .= "\n" . '-- Session Configuration' . "\n\n";
+
+		// The rest of this is only relevant is session is enabled
+		if( isset( $_SESSION ) ) {
+			$return .= 'Session Name:             ' . esc_html( ini_get( 'session.name' ) ) . "\n";
+			$return .= 'Cookie Path:              ' . esc_html( ini_get( 'session.cookie_path' ) ) . "\n";
+			$return .= 'Save Path:                ' . esc_html( ini_get( 'session.save_path' ) ) . "\n";
+			$return .= 'Use Cookies:              ' . ( ini_get( 'session.use_cookies' ) ? 'On' : 'Off' ) . "\n";
+			$return .= 'Use Only Cookies:         ' . ( ini_get( 'session.use_only_cookies' ) ? 'On' : 'Off' ) . "\n";
+		}
+
+		$return  = apply_filters( 'convertkit_sysinfo_after_session_config', $return );
+
+		$return .= "\n" . '### End System Info ###';
+
+		return $return;
+	}
+
+	/**
+	 * Get user host
+     *
+     * Adapted from Easy Digital Downloads
+	 *
+	 * Returns the webhost this site is using if possible
+	 *
+	 * @since 1.6.4
+	 * @return mixed string $host if detected, false otherwise
+	 */
+	function get_host() {
+		$host = false;
+
+		if( defined( 'WPE_APIKEY' ) ) {
+			$host = 'WP Engine';
+		} elseif( defined( 'PAGELYBIN' ) ) {
+			$host = 'Pagely';
+		} elseif( DB_HOST == 'localhost:/tmp/mysql5.sock' ) {
+			$host = 'ICDSoft';
+		} elseif( DB_HOST == 'mysqlv5' ) {
+			$host = 'NetworkSolutions';
+		} elseif( strpos( DB_HOST, 'ipagemysql.com' ) !== false ) {
+			$host = 'iPage';
+		} elseif( strpos( DB_HOST, 'ipowermysql.com' ) !== false ) {
+			$host = 'IPower';
+		} elseif( strpos( DB_HOST, '.gridserver.com' ) !== false ) {
+			$host = 'MediaTemple Grid';
+		} elseif( strpos( DB_HOST, '.pair.com' ) !== false ) {
+			$host = 'pair Networks';
+		} elseif( strpos( DB_HOST, '.stabletransit.com' ) !== false ) {
+			$host = 'Rackspace Cloud';
+		} elseif( strpos( DB_HOST, '.sysfix.eu' ) !== false ) {
+			$host = 'SysFix.eu Power Hosting';
+		} elseif( strpos( $_SERVER['SERVER_NAME'], 'Flywheel' ) !== false ) {
+			$host = 'Flywheel';
+		} else {
+			// Adding a general fallback for data gathering
+			$host = 'DBH: ' . DB_HOST . ', SRV: ' . $_SERVER['SERVER_NAME'];
+		}
+
+		return $host;
+	}
```

Comment:
What is this thing? Is it needed?

Reply:
Ok that's great. Just making sure it was not blindly copied. I see a couple of issues with the code with respect with how we do things in Ruby and I'll note them here, but feel free to merge without making any changes. I just want to write them down as a start of a discussion.

- All those string comparison should use constant instead of inline strings;
- Multiple if/elseif usually is a code smell in Ruby, it could be refactored to not do multiple if/else
- The localhost:/../mysql5.sock is error prone it expects the sock to be named in a very specific way.

Reply:
Awesome. I'm going to merge now so I can get a release out ASAP, but this kind of comment is extremely helpful for me. Thank you 👌